script.js

```diff
@@ -0,0 +1,99 @@
+const display = document.getElementById("display-value");
+const buttons = document.querySelectorAll(".btn");
+
+let currentValue = "0";
+let firstValue = null;
+let operator = null;
+let waitingForSecondValue = false;
+
+function updateDisplay() {
+    display.textContent = currentValue;
+}
+updateDisplay();
+
+buttons.forEach(button => {
+    button.addEventListener('click', (e) => {
+        const value = e.target.dataset.value;
+
+        if (isNumber(value)) {
+            handleNumber(value);
+        } 
+        else if (isOperator(value)) {
+            handleOperator(value);
+        }
+        else if (value === '=') {
+            handleEqual();
+        }
+        else if (value === 'C') {
+            resetCalculator();
+        }
+
+        updateDisplay();
+    });
+});
+
+function isNumber(val) {
+    return !isNaN(val);
+}
```

Comment:
You could add some extra validation logic with this approach:
```suggestion
function isNumber(val) {
    return !isNaN(val) && val !== null && val !== undefined && val !== '';
}
```